src/FhirLoader.Common/FhirResourceClient.cs

```diff
@@ -44,20 +44,22 @@ public BundleClient(string baseUrl, ILogger logger, string? tenantId = null, Asy
             _resiliencyStrategy = resiliencyStrategy ?? DefaultResiliencyStrategy();
         }
 
-        public async Task Send(ProcessedBundle bundle, Action<int, long>? metricsCallback = null, CancellationToken? cancel = null)
+        public async Task Send(ProcessedResource bundle, Action<int, long>? metricsCallback = null, CancellationToken? cancel = null)
         {
-            var content = new StringContent(bundle.BundleText!, Encoding.UTF8, "application/json");
+            var content = new StringContent(bundle.ResourceText!, Encoding.UTF8, "application/json");
             HttpResponseMessage response;
 
+            var requestUri = !string.IsNullOrEmpty(bundle.ResourceType) ? $"/{bundle.ResourceType}" : string.Empty;
+
             var timer = new Stopwatch();
             timer.Start();
 
             try
             {
-                _logger.LogTrace($"Sending {bundle.BundleCount} resources to {_client.BaseAddress}...");
+                _logger.LogTrace($"Sending {bundle.ResourceCount} resources to {_client.BaseAddress}...");
 
                 response = await _resiliencyStrategy.ExecuteAsync(
-                    async ct => await _client.PostAsync("", content, ct),
+                    async ct => await _client.PostAsync(requestUri, content, ct),
```

Comment:
This may need to be a `PUT` following the below logic:
- If the current object being sent is a bundle, `POST`
- If the current object is not a bundle, but doesn't have an `id` property in the JSON, `POST`
- If the current object is not a bundle, but *DOES* have an `id` property in the JSON, `PUT`

Reply:
If I consider the scenirio for ndjson, then it works, but for the package third point, it won't work 
because in profile jsons we do have an id value (e.g us-core-client,us-core-allergyintolerance-clinical-status) without any guid. So in this case, it will always post. 
So bundle and ndjson have this implementation already, where in the handler it checks the id and defines the method post or put.

Reply:
@mikaelweave Logic has been changed to handle the above cases in the FhirResourceClient.cs file. Please verify and share your thoughts.

Reply:
Looks great @vmsuthar56!